conformance_tests/core/test_device/src/test_device.cpp

```diff
@@ -380,6 +380,143 @@ TEST(zeDeviceCanAccessPeerTests,
   EXPECT_EQ(a2b, b2a);
 }
 
+// Return false if uuuid's are NOT equal.
+bool areDeviceUuidsEqual(ze_device_uuid_t uuid1, ze_device_uuid_t uuid2) {
+  if (std::memcmp(&uuid1, &uuid2, sizeof(ze_device_uuid_t))) {
+    return false;
+  }
+  return true;
+}
+
+TEST(zeDeviceCanAccessPeerTests,
+     GivenTheSameDevicesWhenCheckingAccessThenTrueReturned) {
+  auto drivers = lzt::get_all_driver_handles();
+  ASSERT_GT(drivers.size(), 0)
+      << "no drivers found for peer to peer device test";
+
+  std::vector<ze_device_handle_t> all_devices;
+  for (auto driver : drivers) {
+    auto devices = lzt::get_ze_devices(driver);
+    all_devices.insert(all_devices.end(), devices.begin(), devices.end());
+  }
+
+  if (all_devices.size() < 2) {
+    LOG_INFO << "WARNING: Exiting as no multiple devices was found";
+    GTEST_SKIP();
+  }
```

Comment:
This test should be possible to run on single device. The point is to check if the same handle passed as first and second argument returns true.